chunk_downloader.go

```diff
@@ -664,3 +684,32 @@ func copyChunkStream(body io.Reader, rows chan<- []*string) error {
 	}
 	return nil
 }
+
+// A ResultBatch object encapsulates a function that retrieves a subset of rows in a result set.
+type ResultBatch struct {
+	Rec                *[]array.Record
+	idx                int
+	scd                *snowflakeChunkDownloader
+	funcDownloadHelper func(context.Context, *snowflakeChunkDownloader, int) error
+}
+
+// Fetch returns an array of records representing a chunk in the query
+func (rb *ResultBatch) Fetch() error {
+	if rb.Rec != nil {
+		return nil
+	}
+	err := rb.funcDownloadHelper(context.Background(), rb.scd, rb.idx)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
no need to test `err` just `return rb.funcDownloadHelper(context.Background(), rb.scd, rb.idx)`